weather.py

```diff
@@ -0,0 +1,182 @@
+import argparse
+import calendar
+import csv
+import os
+
+required_labels = [
+    "PKT", "PKST", "Max TemperatureC", "Min TemperatureC", "Mean Humidity"
+]
+
+
+def read_weather_file(weather_file_path):
+    monthly_weather_data = []
+    if not os.path.isfile(weather_file_path):
+        return monthly_weather_data
+
+    with open(weather_file_path, mode="r") as file:
+        csv_reader = csv.DictReader(file)
+        headers = [header.strip() for header in csv_reader.fieldnames]
+        for daily_weather_record in csv_reader:
+            daily_weather_record = {
+                key.strip(): value
+                for key, value in daily_weather_record.items()
+            }
+            filtered_daily_record = {
+                key: daily_weather_record[key]
+                for key in required_labels
+                if key in headers
+            }
+            monthly_weather_data.append(filtered_daily_record)
+
+    return monthly_weather_data
+
+
+def load_yearly_weather_data(directory_path, year):
+    year_weather_data = []
+
+    for month in range(1, 13):

```

Comment:
Why we are hardcoding indexes to generate month name by ranges? Can we explore `calendar` library?